src/Controllers/DatabaseController.js

```diff
@@ -616,33 +616,35 @@ DatabaseController.prototype.reduceInRelation = function(className, query, schem
   }
 
   const promises = Object.keys(query).map((key) => {
-    if (query[key] && (query[key]['$in'] || query[key]['$ne'] || query[key]['$nin'] || query[key].__type == 'Pointer')) {
+    if (query[key] == null || query[key] && (query[key]['$in'] || query[key]['$ne'] || query[key]['$nin'] || query[key].__type == 'Pointer')) {
       const t = schema.getExpectedType(className, key);
       if (!t || t.type !== 'Relation') {
         return Promise.resolve(query);
       }
       // Build the list of queries
-      const queries = Object.keys(query[key]).map((constraintKey) => {
-        let relatedIds;
-        let isNegation = false;
-        if (constraintKey === 'objectId') {
-          relatedIds = [query[key].objectId];
-        } else if (constraintKey == '$in') {
-          relatedIds = query[key]['$in'].map(r => r.objectId);
-        } else if (constraintKey == '$nin') {
-          isNegation = true;
-          relatedIds = query[key]['$nin'].map(r => r.objectId);
-        } else if (constraintKey == '$ne') {
-          isNegation = true;
-          relatedIds = [query[key]['$ne'].objectId];
-        } else {
-          return;
-        }
-        return {
-          isNegation,
-          relatedIds
-        }
-      });
+      const queries = query[key] == null ?
```

Comment:
We should probably break that on more lines (as we do for the first early exit), quite tough to read.